core/domain/exp_services.py

```diff
@@ -421,6 +544,20 @@ def export_to_zip_file(exploration_id, version=None):
 
 
 def convert_state_dict_to_yaml(state_dict, width):
+    '''Converting state dict to yaml format
+
+    Args:
+        state_dict: A python dict.
+        width: width(length) for yaml format.
+
+    Returns:
+        State_dict in yaml format.
+
+    Raises:
+        Exception: If state_dict could not be converted into YAML
```

Comment:
No, this exception arises if the state_dict does not represent a valid state. (See line 563; that's where the exception arises.)